### PR TITLE
[Expo Go][Android] Added web socket controls for the CLI

### PR DESCRIPTION
**Pull request by EvanBacon**
# Why

- Add the ability to open the dev menu, reload an experience, and toggle element inspector, etc. from the CLI.
- Related issue https://github.com/expo/expo-cli/issues/160
- iOS PR https://github.com/expo/expo/pull/12151
- CLI PR https://github.com/expo/expo-cli/pull/3280

# How

- Register socket commands for `reload`, `devMenu`, and `sendDevCommand` namespaces. 
- `sendDevCommand` has subcommands:
  - `reload`
  - `toggleDevMenu`
  - `toggleElementInspector`
  - `togglePerformanceMonitor`
  - `toggleRemoteDebugging`

# Test Plan

- [See CLI PR](https://github.com/expo/expo-cli/pull/3280)
- After the changes are approved, I'll backdate them to older versions.

### CODE REVIEW

File: android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java [hunk 1]
```diff
@@ -317,6 +317,18 @@ public void onNewIntent(Intent intent) {
...
+  /**
+   * Handles command line command `adb shell input keyevent 82` that toggles the dev menu on the current experience activity.
+   */
+  public boolean toggleDevMenu() {
+    if (mReactInstanceManager != null && mReactInstanceManager.isNotNull() && !mIsCrashed) {
+      mDevMenuManager.toggleInActivity(this);
+      return true;
+    }
+    return false;
+  }
```
Comment (lukmccall) [inline android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java:329]: Is this method called when the user runs `adb shell input keyevent 82`? If no, change the comment ;)

**Review by lukmccall — APPROVED** [review]
LGTM 💪

File: android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java [hunk 2]
```diff
@@ -317,6 +317,18 @@ public void onNewIntent(Intent intent) {
     }
   }
 
+  /**
+   * Handles command line command `adb shell input keyevent 82` that toggles the dev menu on the current experience activity.
+   */
```
Comment (EvanBacon) [inline android/expoview/src/main/java/host/exp/exponent/experience/ExperienceActivity.java:322]:
```suggestion
```

File: android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.java [hunk 3]
```diff
@@ -3,22 +3,176 @@
...
+      // Get permission to show debug overlay in dev builds.
+      if (!Settings.canDrawOverlays(context)) {
+        Intent intent =
+          new Intent(
+            Settings.ACTION_MANAGE_OVERLAY_PERMISSION,
+            Uri.parse("package:" + context.getPackageName()));
+        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        FLog.w(
+          ReactConstants.TAG,
+          "Overlay permissions needs to be granted in order for react native apps to run in dev mode");
```
Comment (brentvatne) [inline android/expoview/src/main/java/versioned/host/exp/exponent/VersionedUtils.java:80]:
```suggestion
          "Overlay permissions needs to be granted in order for React Native apps to run in development mode");
```

**Review by brentvatne — APPROVED** [review]

**Pull request merged**